yt_dlp/extractor/common.py

```diff
@@ -1188,13 +1188,16 @@ def _search_regex(self, pattern, string, name, default=NO_DEFAULT, fatal=True, f
             self.report_warning('unable to extract %s' % _name + bug_reports_message())
             return None
 
-    def _search_json(self, start_pattern, string, name, video_id, *, end_pattern='', contains_pattern='.+', fatal=True, **kwargs):
+    def _search_json(self, start_pattern, string, name, video_id, *, end_pattern='',
+                     contains_pattern='.+', fatal=True, default=NO_DEFAULT, **kwargs):
         """Searches string for the JSON object specified by start_pattern"""
         # NB: end_pattern is only used to reduce the size of the initial match
         return self._parse_json(
-            self._search_regex(rf'{start_pattern}\s*(?P<json>{{{contains_pattern}}})\s*{end_pattern}',
-                               string, name, group='json', fatal=fatal) or '{}',
-            video_id, fatal=fatal, ignore_extra=True, **kwargs) or {}
+            self._search_regex(
+                rf'{start_pattern}\s*(?P<json>{{{contains_pattern}}})\s*{end_pattern}',
+                string, name, group='json', fatal=fatal,
+                default='{}' if default is not NO_DEFAULT else NO_DEFAULT),
+            video_id, fatal=fatal, ignore_extra=True, **kwargs) or {} if default is NO_DEFAULT else default
```

Comment:
I don't think this behaves correctly. Pls double check

What we want:
* `fatal=True` => throw regex error
* `fatal=False` => print regex warning, but not json warning and return `{}`
* `default=xxx` => print neither warning and return `xxx` (not `json.loads(xxx)`)

Note that `fatal` should be ignored in the last case

(split into 2 statements if needed)

You can merge once you confirm this behavior

Reply:
hmm ok. I've had a go at fixing it up, you might want to have a look to see if you can simplify it.

would add a test but not sure if we have an easy way for testing warnings